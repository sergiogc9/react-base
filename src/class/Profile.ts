export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
