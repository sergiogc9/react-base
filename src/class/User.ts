export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
