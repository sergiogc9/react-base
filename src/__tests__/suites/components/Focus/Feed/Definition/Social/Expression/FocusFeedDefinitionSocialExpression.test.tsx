import React from "react";
import { shallow, mount } from "enzyme";

import FocusFeedDefinitionSocialExpression from "@src/components/Focus/Feed/Definition/Social/Expression/FocusFeedDefinitionSocialExpression";
import { ScopeSocial, SocialInstagramAccount, DefinitionSocialInstagramAccount } from "@src/class/Feed";

const query = "kketa";
const enabled = false;
const scope: ScopeSocial[] = ['tags'];
const expressionKey = "include_expressions";

const instagramAccount: SocialInstagramAccount = {
	id: "id1",
	name: "Girona FC",
	profile_picture_url: "fakeurl.com",
	screen_name: "gironafc",
	valid: true
};
const instagramAccounts: SocialInstagramAccount[] = [instagramAccount];
const mainDefinitionInstagramAccount: DefinitionSocialInstagramAccount = {
	id: "id1",
	screen_name: "gironafc",
	linkedExpression: { type: "main", index: 0 }
};
const includeDefintionInstagramAccount: DefinitionSocialInstagramAccount = {
	id: "id1",
	screen_name: "gironafc",
	linkedExpression: { type: "include_expressions", index: 0 }
};

describe("Definition social expression", () => {
	let wrapper: any;
	let onChangeQueryMock = jest.fn();
	let onCheckQueryMock = jest.fn();
	let onChangeScopeMock = jest.fn();
	let onToggleEnableQueryMock = jest.fn();
	let onRemoveExpressionMock = jest.fn();

	const getComponent = () => (
		<FocusFeedDefinitionSocialExpression
			expressionKey={expressionKey}
			index={0}
			query={query}
			enabled={enabled}
			scope={scope}
			instagramAccounts={instagramAccounts}
			definitionInstagramAccounts={[]}
			onQueryChanged={onChangeQueryMock}
			onCheckQuery={onCheckQueryMock}
			onEnabledToggled={onToggleEnableQueryMock}
			onScopeChanged={onChangeScopeMock}
			onRemoveExpression={onRemoveExpressionMock}
		/>
	);

	beforeEach(() => {
		onChangeQueryMock = jest.fn();
		onCheckQueryMock = jest.fn();
		onChangeScopeMock = jest.fn();
		onToggleEnableQueryMock = jest.fn();
		onRemoveExpressionMock = jest.fn();
		wrapper = shallow(getComponent());
	});

	it("Component default snaphsot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component default snaphsot with error", () => {
		wrapper.setProps({ error: true });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component default snaphsot being main expression", () => {
		wrapper.setProps({ expressionKey: "main" });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component default snaphsot with query enabled", () => {
		wrapper.setProps({ enabled: true });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("On toggle enabled query handler", () => {
		expect(onToggleEnableQueryMock).toHaveBeenCalledTimes(0);
		wrapper.find(`#focusFeedDefinitionSocialExpressionEnabled${expressionKey}0`).at(0).props().onChange(false);
		expect(onToggleEnableQueryMock).toHaveBeenCalledTimes(1);
		expect(onToggleEnableQueryMock).toHaveBeenCalledWith(false);
	});

	it("On change query handler", () => {
		expect(onChangeQueryMock).toHaveBeenCalledTimes(0);
		wrapper.find(`TextField#focusFeedSocialExpressionQuery${expressionKey}0`).at(0).props().onChange('queryy');
		expect(onChangeQueryMock).toHaveBeenCalledTimes(1);
		expect(onChangeQueryMock).toHaveBeenCalledWith("queryy");
	});

	it("On blur query query handler", () => {
		expect(onCheckQueryMock).toHaveBeenCalledTimes(0);
		wrapper.find(`TextField#focusFeedSocialExpressionQuery${expressionKey}0`).at(0).props().onBlur();
		expect(onCheckQueryMock).toHaveBeenCalledTimes(1);
		expect(onCheckQueryMock).toHaveBeenCalledWith();
	});

	it("On toggle enabled query handler", () => {
		expect(onChangeScopeMock).toHaveBeenCalledTimes(0);
		wrapper.find(`#focusFeedSocialDefinitionExpressionScope${expressionKey}0`).at(0).props().onChange(['tags']);
		expect(onChangeScopeMock).toHaveBeenCalledTimes(1);
		expect(onChangeScopeMock).toHaveBeenCalledWith(['tags']);
	});

	it("On remove expression", () => {
		expect(onRemoveExpressionMock).toHaveBeenCalledTimes(0);
		wrapper.find('.definition-expression-remove-button').at(0).simulate('click');
		expect(onRemoveExpressionMock).toHaveBeenCalledTimes(1);
	});

	it("On remove unremovible expression", () => {
		wrapper = shallow(<FocusFeedDefinitionSocialExpression
			expressionKey={expressionKey}
			index={0}
			query={query}
			enabled={enabled}
			scope={scope}
			onQueryChanged={onChangeQueryMock}
			onCheckQuery={onCheckQueryMock}
			onEnabledToggled={onToggleEnableQueryMock}
			onScopeChanged={onChangeScopeMock}
			instagramAccounts={instagramAccounts}
			definitionInstagramAccounts={[]}
		/>);
		expect(onRemoveExpressionMock).toHaveBeenCalledTimes(0);
		wrapper.find('.definition-expression-remove-button').at(0).simulate('click');
		expect(onRemoveExpressionMock).toHaveBeenCalledTimes(0);
	});

	it("Component snapshot with selected instagram account", () => {
		wrapper.setProps({ definitionInstagramAccounts: [includeDefintionInstagramAccount] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Selected instagram account on clicked handler", () => {
		const onInstagramAccountToggled = jest.fn();
		wrapper.setProps({ definitionInstagramAccounts: [includeDefintionInstagramAccount], onInstagramAccountToggled });
		expect(onInstagramAccountToggled).toHaveBeenCalledTimes(0);
		wrapper.find('FocusFeedDefinitionSocialInstagramAccount').at(0).props().onSelected();
		expect(onInstagramAccountToggled).toHaveBeenCalledTimes(1);
		expect(onInstagramAccountToggled).toHaveBeenCalledWith({
			id: includeDefintionInstagramAccount.id,
			screen_name: includeDefintionInstagramAccount.screen_name,
			linkedExpression: { type: expressionKey, index: 0 }
		});
	});

	it("Component snapshot with selected instagram account but without instagram accounts loaded", () => {
		wrapper.setProps({ instagramAccounts: null, definitionInstagramAccounts: [includeDefintionInstagramAccount] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot showing instagram accounts list", () => {
		wrapper.setProps({ scope: ["mentions"] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Focus query input if query is empty", () => {
		// TODO FIX!
		const component = mount(<FocusFeedDefinitionSocialExpression
			expressionKey={expressionKey}
			index={0}
			query={""}
			enabled={enabled}
			scope={scope}
			instagramAccounts={instagramAccounts}
			definitionInstagramAccounts={[]}
			onQueryChanged={onChangeQueryMock}
			onCheckQuery={onCheckQueryMock}
			onEnabledToggled={onToggleEnableQueryMock}
			onScopeChanged={onChangeScopeMock}
			onRemoveExpression={onRemoveExpressionMock}
		/>);
		expect(component.html()).toMatchSnapshot();
	});
});
