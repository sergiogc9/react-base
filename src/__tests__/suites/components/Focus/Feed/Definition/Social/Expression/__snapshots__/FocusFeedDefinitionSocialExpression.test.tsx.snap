// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition social expression Component default snaphsot 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"tags\\" disabled=\\"\\"/></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input definition-expressions-query \\"><input type=\\"text\\" value=\\"kketa\\" id=\\"focusFeedSocialExpressionQueryinclude_expressions0\\" disabled=\\"\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;

exports[`Definition social expression Component default snaphsot being main expression 1`] = `"<div class=\\"social-definition-expression content-expression main-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledmain0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledmain0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledmain0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledmain0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.label.main_expression_social_media_react</span><div id=\\"focusFeedSocialDefinitionExpressionScopemain0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopemain0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopemain0\\" value=\\"tags\\" disabled=\\"\\"/></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input definition-expressions-query \\"><input type=\\"text\\" value=\\"kketa\\" id=\\"focusFeedSocialExpressionQuerymain0\\" disabled=\\"\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div></div>"`;

exports[`Definition social expression Component default snaphsot with error 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"tags\\" disabled=\\"\\"/></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input definition-expressions-query wrong-query\\"><input type=\\"text\\" value=\\"kketa\\" id=\\"focusFeedSocialExpressionQueryinclude_expressions0\\" disabled=\\"\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--text-field-error\\"/></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;

exports[`Definition social expression Component default snaphsot with query enabled 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression enabled\\"><div class=\\"md-selection-control-container definition-expression-enabled checkbox-green \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"tags\\"/></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--input definition-expressions-query \\"><input type=\\"text\\" value=\\"kketa\\" id=\\"focusFeedSocialExpressionQueryinclude_expressions0\\" class=\\"md-text-field md-text-field--margin md-text\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;

exports[`Definition social expression Component snapshot showing instagram accounts list 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.mention_react</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"mentions\\" disabled=\\"\\"/></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input definition-expressions-query \\"><input type=\\"text\\" value=\\"kketa\\" id=\\"focusFeedSocialExpressionQueryinclude_expressions0\\" disabled=\\"\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button><div id=\\"focusFeedSocialDefinitionInstagramAccountsinclude_expressions0\\" class=\\"social-definition-instagram-accounts-list\\"><div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span></div><div id=\\"focusFeedDefinitionSocialInstagramAccountAddNew\\" class=\\"instagram-account-add\\"><i class=\\"md-icon material-icons\\">add</i><span>feed.form.instagram.legend.button_react</span></div></div></div>"`;

exports[`Definition social expression Component snapshot with selected instagram account 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"tags\\" disabled=\\"\\"/></div></div></div><div id=\\"focusFeedDefinitionSocialInstagramAccountid1\\" class=\\"instagram-account \\"><img src=\\"fakeurl.com\\" alt=\\"\\"/><span>Girona FC</span><span>(@gironafc)</span><i class=\\"md-icon material-icons\\">clear</i></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;

exports[`Definition social expression Component snapshot with selected instagram account but without instagram accounts loaded 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"/><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" value=\\"tags\\" disabled=\\"\\"/></div></div></div><div></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;

exports[`Definition social expression Focus query input if query is empty 1`] = `"<div class=\\"social-definition-expression content-expression is-first-expression disabled\\"><div class=\\"md-selection-control-container definition-expression-enabled  \\"><input id=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" aria-hidden=\\"true\\" value=\\"\\"><label for=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialExpressionEnabledinclude_expressions0\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"definition-expression-label\\">feed.form.or</span><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field definition-expression-scope\\"><div class=\\"md-select-field__toggle\\"><div id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0-toggle\\" role=\\"listbox\\" disabled=\\"\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-fake-btn--no-outline md-select-field md-text--disabled\\"><div class=\\"md-icon-separator md-text-field md-text-field--margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">feed.form.scope.option.tag</span><i class=\\"md-icon material-icons md-text--disabled\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"focusFeedSocialDefinitionExpressionScopeinclude_expressions0\\" disabled=\\"\\" value=\\"tags\\"></div></div></div><div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input definition-expressions-query \\"><input id=\\"focusFeedSocialExpressionQueryinclude_expressions0\\" disabled=\\"\\" type=\\"text\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div><button id=\\"focusFeedSocialExpressionRemoveinclude_expressions0\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block definition-expression-remove-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">delete</i></button></div>"`;
